src/main/java/org/zalando/nakadi/config/RepositoriesConfig.java

```diff
@@ -13,28 +15,37 @@
 import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
 import org.zalando.nakadi.repository.zookeeper.ZookeeperConfig;
 import org.zalando.nakadi.service.timeline.TimelineSync;
-import org.zalando.nakadi.util.FeatureToggleService;
-import org.zalando.nakadi.util.FeatureToggleServiceDefault;
-import org.zalando.nakadi.util.FeatureToggleServiceZk;
+import org.zalando.nakadi.service.FeatureToggleService;
+import org.zalando.nakadi.service.FeatureToggleServiceZk;
 import org.zalando.nakadi.validation.EventBodyMustRespectSchema;
 import org.zalando.nakadi.validation.EventMetadataValidationStrategy;
 import org.zalando.nakadi.validation.JsonSchemaEnrichment;
 import org.zalando.nakadi.validation.ValidationStrategy;
 
+import java.util.Set;
+
 @Configuration
 @Profile("!test")
 @Import({KafkaConfig.class, ZookeeperConfig.class})
 public class RepositoriesConfig {
 
+    private static final Logger LOG = LoggerFactory.getLogger(RepositoriesConfig.class);
+
     @Bean
     public FeatureToggleService featureToggleService(
             @Value("${nakadi.featureToggle.default}") final boolean forceDefault,
-            final ZooKeeperHolder zooKeeperHolder) {
+            final ZooKeeperHolder zooKeeperHolder, final FeaturesConfig featuresConfig) {
+        final FeatureToggleService featureToggleService = new FeatureToggleServiceZk(zooKeeperHolder);
         if (forceDefault) {
-            return new FeatureToggleServiceDefault();
-        } else {
-            return new FeatureToggleServiceZk(zooKeeperHolder);
+            final Set<String> features = featuresConfig.getDefaultFeatures().keySet();
+            for (final String feature: features) {
+                LOG.info("Setting feature {} to {}", feature, featuresConfig.getDefault(feature));
```

Comment:
hard to get the naming `getDefault`, what do you think to rename it to `isEnabled`, `getDefaultState` etc..

Reply:
I like `getDefaultState`